solr/core/src/java/org/apache/solr/util/circuitbreaker/CPUCircuitBreaker.java

```diff
@@ -33,39 +35,47 @@
  * We depend on OperatingSystemMXBean which does not allow a configurable interval of collection of
  * data.
  */
-public class CPUCircuitBreaker extends CircuitBreaker {
+public class CPUCircuitBreaker extends CircuitBreaker implements SolrCoreAware {
   private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
 
-  private boolean enabled = true;
+  private Boolean enabled = null;
   private double cpuUsageThreshold;
-  private final SolrCore core;
+  private CoreContainer cc;
 
   private static final ThreadLocal<Double> seenCPUUsage = ThreadLocal.withInitial(() -> 0.0);
 
   private static final ThreadLocal<Double> allowedCPUUsage = ThreadLocal.withInitial(() -> 0.0);
 
-  public CPUCircuitBreaker(SolrCore core) {
+  public CPUCircuitBreaker() {
     super();
-    this.core = core;
+  }
+
+  public CPUCircuitBreaker(CoreContainer coreContainer) {
+    super();
+    this.cc = coreContainer;
   }
```

Comment:
Wondering if we need to worry about backwards compatibility here, maybe not maintaining it but upgrade-note-explaining the signature change perhaps? `CircuitBreakerRegistry` is tagged as experimental but this class is not i.e. in principle folks could have sub-classed a custom version of it or so.

Reply:
Sure, will keep the old constructor and deprecate it.